src/services/container.ts

```diff
@@ -0,0 +1,19 @@
+import { env } from '@/services/env'
```

Comment:
Its a ServiceContainer a centralized place to access (or inject from The Outside) Services, it'll be swapped out for something a little more featured later (i.e. with auto instantiation, lazy instantiation etc etc)

Reply:
I’d prefer for such a pattern to be justified by a use case that makes use of features as you named them. I’m curious to hear of examples.

In the context of this PR (i.e. streamlining the access of static data for the purpose of avoiding hard-coded docs URLs), this doesn’t seem to be needed.

Reply:
One use case is to avoid merge conflicts between repositories.

Examples of auto instantiation would be the service container doing all the dependency lookup for a service when you first instantiate it, lazy instantiation, as sometimes you don't want to instantiate a service until you need it (here we instantiate Env and add it to the container manually even though we don't necessarily need it yet)

> In the context of this PR (i.e. streamlining the access of static data for the purpose of avoiding hard-coded docs URLs), this doesn’t seem to be needed.

A fair point I guess, but you can do a lot of things differently, but depends if you are thinking short term or long term as how you might go about doing something.

Just to point out, you don't even need to know that it's using a ServiceContainer if you are just using it as a consumer (i.e. `const env = useEnv()`, in fact it looks just like any other Vue composable. If we wanted to change our building tooling again or how the app is fed these variables again, the app doesn't need to know about it, we just tweak the service or inject a new one from The Outside.

Reply:
What you describe may well apply to complex utilities, but it doesn’t apply to this use case.

> One use case is to avoid merge conflicts between repositories.

Would you mind elaborating how that is accomplished for looking up static data? As far as I can tell, you’d get the same merge conflict whether you make changes to a line referencing `env('PROPERTY')` as you’d get for `env.PROPERTY`. The same would be true for the code where these values are defined.

> A fair point I guess, but you can do a lot of things differently, but depends if you are thinking short term or long term as how you might go about doing something.

That’s why I’m asking. Without a service that has sufficient complexity to warrant the feature richness you mentioned, this adds a maintenance burden in the short term and possibly also in the long term (depending on if and when such a service becomes necessary).

> Just to point out, you don't even need to know that it's using a ServiceContainer if you are just using it as a consumer (i.e. const env = useEnv()), in fact it looks just like any other Vue composable. If we wanted to change our build tooling again (or the build tooling itself changed) or change how the app is fed these variables again, the app doesn't need to know about it, we just tweak the service or inject a new one from The Outside.

That’s also true for accessing a plain object with `env.PROPERTY`. It’s plain JavaScript and not framework or tooling specific. The things that would change on the `env` utility are (1) new properties being added, (2) properties being removed, or (3) properties being renamed. All three scenarios have the exact same impact regardless of whether they’re accessed directly from an object or via a service abstraction.

Reply:
It was decided to move forward with this implementation because it works and because moving between the approaches in question is easy.

(For the record, I still don’t think it’s warranted to introduce this kind of abstraction for such simple utilities.)